MULTIARCH-4515: add new endpoint to receive initrd.addrsize (#199)
openshift/assisted-image-service: README.md

- `version`: indicates the version of the RHCOS base image to use (must match an entry in `RHCOS_VERSIONS`)
- `arch`: the base image cpu architecture (must match an entry in `RHCOS_VERSIONS`)

> - `addrsize`: indicates if the initrd.addrsize (`1`) file instead of initrd.img (`0`) should be returned. Only valid for s390x archtiecture

Author: @paul-maidment Thank you for bringing this up. Maybe I bring some light into the thoughts from s390x community. Values like 0 and 1 are very common. In boot files like parm files values rd.neednet=1 or zfcp.allow_lun_scan=0 are used. Therefor I choose the value that way. But I understand that for users working with different architectures would get confused.

So what if we do it that way:
- change the parameter from 'addrsize' to 'file'
- value initrd.img or initrrd.addrsize with default initrd.img

Is this a feasible way to go?

> - `addrsize`: indicates if the initrd.addrsize (`1`) file instead of initrd.img (`0`) should be returned. Only valid for s390x archtiecture

Reviewer: It should just be a boolean. true or false

> - `addrsize`: indicates if the initrd.addrsize (`1`) file instead of initrd.img (`0`) should be returned. Only valid for s390x archtiecture

Reviewer: It should just be a boolean. true or false

> - `addrsize`: indicates if the initrd.addrsize (`1`) file instead of initrd.img (`0`) should be returned. Only valid for s390x archtiecture

Reviewer: See below suggestion for my recommended phrasing. I mostly didn't like the use of "returned" since it's a too generic word and doesn't give much context of when and what

> - `addrsize`: indicates if the initrd.addrsize (`1`) file instead of initrd.img (`0`) should be returned. Only valid for s390x archtiecture

Reviewer: >from s390x community. Values like 0 and 1 are very common

This is the semantics of our HTTP service API, the conventions of the s390x community / kernel boot arguments don't apply

- `Authorization`: this header is passed directly through to assisted service requests to handle RHSSO authentication

> ### `GET /images/{image_id}/initrd-addrsize`

Reviewer: I would like to emphasize how specific this is, so let's have it be part of the name

```suggestion
### `GET /images/{image_id}/s390x-initrd-addrsize`
```

Reviewer: Or are there other IBM architectures which this might be relevant to in the future?

Author: No yet ... so for the time being s390x only.

Author: Done

> Downloads the initrd.addrsize containing the psw of the initrd (8 bytes) and the size of the initrd (8 bytes). This if for s390x architecture only.

Reviewer: ```suggestion
Downloads the initrd.addrsize containing the psw of the initrd (8 bytes) and the size of the initrd (8 bytes). This is for s390x architecture only.
```

Author: Done

#### Query parameters

- `version`: indicates the version of the RHCOS base image to use (must match an entry in `RHCOS_VERSIONS`)

> - `arch`: the base image cpu architecture (must match an entry in `RHCOS_VERSIONS`)

Reviewer: ```suggestion
- `arch`: the base image cpu architecture (for this endpoint, this must be set to s390x)
```

Author: Done

> - `arch`: the base image cpu architecture (for this endpoint, this must be set to s390x)

Author: Done
